dateparser/search/search.py

```diff
@@ -86,10 +86,12 @@ def split_if_not_parsed(self, item, original):
                 possible_splits.extend(self.split_by(item, original, splitter))
         return possible_splits
 
-    def parse_item(self, parser, item, translated_item, parsed, need_relative_base):
+    def parse_item(self, parser, item, translated_item, parsed, need_relative_base, language):
         relative_base = None
-        item = item.replace('ngày', '')
-        item = item.replace('am', '')
+
+        if language == "de":
```

Comment:
For the German language making an edge case

Reply:
This seems like a good fix. In case of Vietnamese, I would add:

```python
if language == "vi":
    item = item.replace('ngày', '')
```

I don't know the reasons, but I would prefer to keep it there as it was probably because of some reason, so I would prefer to keep it.

When it was introduced is here: https://github.com/scrapinghub/dateparser/commit/7e7e22417e71e103225a2fdf05e68a6e67db041c#diff-4e4813a9f7a6e85270fb6054ada722c2a230066dc9fc9d442250b0da38acbfd0R26